internal/schema/schema_test.go

```diff
@@ -0,0 +1,370 @@
+				t.Fail()
+			}
+
+			if validationErrorList.Errors != nil && !tc.wantErr {
+				t.Log("wanted no error, but error occurred")
+				t.Fail()
+			} else if validationErrorList.Errors == nil && tc.wantErr {
+				t.Log("wanted error, but no error present")
+				t.Fail()
+			}
+		}
+	})
+}
+
+func TestMethods(t *testing.T) {
+	t.Run("schema.Manager.walkInputProperties", func(t *testing.T) {
+		mgr := &Manager{}
+
+		var testCases = []struct {
+			name    string
+			input   map[string]*structpb.Value
+			want    map[string]string
+			wantErr bool
+		}{
+			{
+				name: "Test 1",
+				input: map[string]*structpb.Value{
+					"somekey": {
+						Kind: &structpb.Value_StructValue{
+							StructValue: &structpb.Struct{
+								Fields: map[string]*structpb.Value{
+									"somekey1": {
+										Kind: &structpb.Value_StringValue{
+											StringValue: "somevalue1",
+										},
+									},
+									"somekey2": {
+										Kind: &structpb.Value_NumberValue{
+											NumberValue: 2,
+										},
+									},
+								},
+							},
+						},
+					},
+					"somekey3": {
+						Kind: &structpb.Value_BoolValue{
+							BoolValue: true,
+						},
+					},
+				},
+				want: map[string]string{
+					"/somekey/somekey1": "/somekey/somekey1",
+					"/somekey/somekey2": "/somekey/somekey2",
+					"/somekey3":         "/somekey3",
+				},
+				wantErr: false,
+			},
+			{
+				name:    "Test 2 - Nil Check",
+				input:   nil,
+				want:    nil,
+				wantErr: false,
+			},
+		}
+
+		for _, tc := range testCases {
+			var have = make(map[string]string)
+			mgr.walkInputProperties("", tc.input, have)
+
+			eq := mapsEqual(tc.want, have)
+			if !eq && !tc.wantErr {
+				t.Log("wanted map is not equal to resulting map")
+				t.Fail()
+			} else if eq && tc.wantErr {
+				t.Log("wanted map is equal to resulting map, but error expected")
+				t.Fail()
+			}
+		}
+	})
+
+	t.Run("schema.Manager.walkSchemaProperties", func(t *testing.T) {
+		mgr := &Manager{}
+
+		var testCases = []struct {
+			name          string
+			inputSchema   string
+			wantPrincipal map[string]string
+			wantResource  map[string]string
+			wantErr       bool
+		}{
+			{
+				name:        "Test 1",
+				inputSchema: "LS0tCmFwaVZlcnNpb246ICJhcGkuY2VyYm9zLmRldi92MSIKZGVzY3JpcHRpb246IHwtCiBTY2hlbWEgZGVmaW5pdGlvbiBmaWxlCgpwcmluY2lwYWxTY2hlbWE6ICMgZGVmaW5lIHNjaGVtYSBmb3IgcmVxdWVzdC5wcmluY2lwYWwuYXR0cgogIHR5cGU6IG9iamVjdAogIHByb3BlcnRpZXM6CiAgICBwZXJtaXNzaW9uczoKICAgICAgdHlwZTogYXJyYXkKICAgICAgZGVzY3JpcHRpb246ICJwZXJtaXNzaW9ucyBkZWZpbmVkIGZvciB0aGlzIHVzZXIiCiAgICBlZGl0b3JfZm9yX2NhdGVnb3JpZXM6CiAgICAgIHR5cGU6IGFycmF5CiAgICBwZXJzb25hbF9kZXRhaWxzOgogICAgICB0eXBlOiBvYmplY3QKICAgICAgcHJvcGVydGllczoKICAgICAgICBuYW1lOgogICAgICAgICAgdHlwZTogc3RyaW5nCiAgICAgICAgbWFpbDoKICAgICAgICAgIHR5cGU6IHN0cmluZwoKcmVzb3VyY2VTY2hlbWFzOiAjIGRlZmluZSBzY2hlbWEgZm9yIHJlcXVlc3QucmVzb3VyY2UuYXR0cgogIHJlc291cmNlX2Jsb2c6CiAgICB0eXBlOiBvYmplY3QKICAgIHByb3BlcnRpZXM6CiAgICAgIGNhdGVnb3JpZXM6CiAgICAgICAgdHlwZTogc3RyaW5nCg==",
```

Comment:
This test is not very maintainable because whoever is looking at this will have to decode this and reencode it. Didn't you introduce templating to the test framework to handle this kind of thing already?